lib/generators/templates/section.yml

```diff
@@ -16,6 +16,36 @@ default: &default
     # highlighting
     hl_num: 5
 
+  facets:
```

Comment:
I'm picturing that we will probably want to do this for the existing facets also, long term, instead of leaving them in the models file.  Is that what you are picturing, also?

Reply:
That's definitely where my thoughts were leaning, yes. Where we put them, I was less certain of. Maybe alongside `api_options` in `public.yml` so the organization kind of matches what I did in the section YAML file?

Reply:
I think that makes the most sense to me.  Then the facets.rb file will only be for the methods that delegate which facets to pick and which returning facet values to translate, etc.
Heck, if we don't want that code in the models, I think there's a compelling argument for it going somewhere else, too, I just don't know....where....

Reply:
I think we should move it all out of models. The code never interacts with a database, so models seems like the wrong place in hindsight.

I think application_controller.rb is the ideal place since the methods will only be called from application_controller.rb itself and items_controller.rb IIRC.

Reply:
I'm thinking this should be a separate branch/PR as that will definitely break all existing apps. As is, this branch is compatible with existing apps :)

Reply:
Going to change app-wide facets to come from a YAML file as well after all :smile: